### PR TITLE
Create longest_path_in_a_matrix.py

### PR DESCRIPTION
**Pull request by akulagrawal**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by akulagrawal** [comment]
@Kush1101 please approve

**Comment by akulagrawal** [comment]
@Kush1101 please approve

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

File: dynamic_programming/longest_path_in_matrix.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    ):
+        w = 1 + find_longest_path_from_a_cell(y_coordinate + 1, x_coordinate, mat, dp)
+
+    dp[y_coordinate][x_coordinate] = max(x, max(y, max(z, max(w, 1))))
+    return dp[y_coordinate][x_coordinate]
+
+
+def find_longest_path_in_a_matrix(mat: list[list[int]]) -> int:
+    """
+    Find the length of longest path of increasing sequence in a matrix mat.
```
Comment (poyea) [inline dynamic_programming/longest_path_in_matrix.py:45]: It's a bit unclear on what the problem is here. Could you provide a reference / explanation for a "longest path"?

**Review by poyea — CHANGES_REQUESTED** [review]
Please add a few references

**Comment by tianyizheng02** [comment]
Closing due to lack of response to code review. Also, I believe this PR is plagiarized from [this GeeksForGeeks article](https://www.geeksforgeeks.org/find-the-longest-path-in-a-matrix-with-given-constraints/) because the code structure is identical with the only differences being function names and some variable names.

**Pull request closed without merging**